tests/saitests/py3/sai_qos_tests.py

```diff
@@ -1636,15 +1640,19 @@ def runTest(self):
                      pkts_num_dismiss_pfc - hysteresis) // cell_occupancy
                 )
 
-            if hwsku == 'Arista-7050CX3-32S-D48C8' or hwsku == 'Arista-7050CX3-32S-C32' or hwsku == 'DellEMC-Z9332f-M-O16C64' or hwsku == 'DellEMC-Z9332f-O32':
-                xmit_counters, queue_counters = sai_thrift_read_port_counters(
-                    self.client, port_list[dst_port_id])
-                actual_port_leak_out = xmit_counters[TRANSMITTED_PKTS] - \
-                    xmit_counters_base[TRANSMITTED_PKTS]
-                send_packet(self, src_port_id, pkt, actual_port_leak_out)
+            if 'broadcom' in asic_type.lower() or hwsku == 'Arista-7050CX3-32S-D48C8' or hwsku == 'Arista-7050CX3-32S-C32' or hwsku == 'DellEMC-Z9332f-M-O16C64' or hwsku == 'DellEMC-Z9332f-O32':
+                xmit_counters, _ = sai_thrift_read_port_counters(self.client, port_list[dst_port_id])
+                counters_base = xmit_counters_base
+                actual_port_leak_out = xmit_counters[TRANSMITTED_PKTS] - counters_base[TRANSMITTED_PKTS]
+                while actual_port_leak_out > 0:
```

Comment:
why there is no delay between the counter check?

Reply:
counter value come from sai, instead of sonic counter db.
so don't need delay.